### PR TITLE
Support testing against PHP 8 and 8.1.

### PR DESCRIPTION
**Pull request by dcr-stripe**
## Summary
This PR:
* Runs CI for PHP 8.0 and PHP 8.1.
* Changes `composer.json` to require (as a devDependency) EITHER phpunit 5.7, which is compatible with PHP <=7.2; OR phpunit 9.0, which is compatible with PHP 8.
  * Some functions in phpunit 9.0 don't exist in phpunit 5.7. To keep our test suite compatible with both, we extend the PHPUnit `TestCase` class and add some "shims" over top of them.
* Fixes the flaky StripeTelemetryTest
* Updates the php-cs-fixer task to run on PHP8
  * Towards that end, upgrade php-cs-fixer to the recent version 3.2.1 (`|| 2.17.1`  is provided as a fallback so composer install doesn't break on old versions of PHP)
  * Make small formatting/style changes associated with that upgrade.

fixes #1200 #1199

### CODE REVIEW

**Comment by richardm-stripe** [comment]
@dcr-stripe will you take a look at this?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,22 +19,28 @@ jobs:
...
   phpstan:
     name: PHPStan
 
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "7.4"
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:35]: Should we run this on 8.1 as well?

File: composer.json [hunk 2]
```diff
@@ -21,11 +21,10 @@
     "ext-mbstring": "*"
   },
   "require-dev": {
-    "phpunit/phpunit": "^5.7",
+    "phpunit/phpunit": "^5.7 || ^9.0",
     "php-coveralls/php-coveralls": "^2.1",
     "squizlabs/php_codesniffer": "^3.3",
-    "symfony/process": "~3.4",
```
Comment (dcr-stripe) [inline composer.json:27]: Confirming this was just a stale require?

**Review by dcr-stripe — COMMENTED** [review]
This looks good to me overall! Thanks for all your hard work on this Richard! 🎉 

(I can't approve since it's my own PR)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (richardm-stripe) [reply .github/workflows/ci.yml:35]: Yes, I have that in a forthcoming PR. There are some warnings I need to address first.

File: composer.json (hunk 2 above)
Comment (richardm-stripe) [reply composer.json:27]: Good call. We actually use it in `StripeMock.php` which apparently attempts to boot up stripe-mock if you aren't already running it?

In our CI we always boot up stripe-mock in a separate step, and I always boot stripe-mock up manually, so I went ahead and removed this file.

Comment (dcr-stripe) [reply composer.json:27]: LGTM - good find!

**Comment by hivokas** [comment]
This is awesome, thanks for your work!

**Pull request merged**